File: c/hash-table/find-the-difference.c
char findTheDifference(char* s, char* t) {
    int hash[256] = {0};
    int len = strlen(s);
    int i;

    for(i = 0; i < len; ++i)
        hash[s[i]]++;

    for(i = 0; i <= len; ++i)
        if(hash[t[i]] == 0)
            break;
        else
            --hash[t[i]];

    return t[i];
}

File: python/hash-table/find-the-difference.py
class Solution(object):
    def findTheDifference(self, s, t):
        h = {};
        for key in s:
            if h.get(key) is None:
                h[key] = 1
            else:
                h[key] += 1

        for key in t:
            if h.get(key,0) == 0:
                return key
            h[key] -= 1
